hydromt/workflows/region.py

```diff
@@ -0,0 +1,293 @@
+
+        Subbasins are defined by its outlet locations and include all area upstream
+        from these points. The outlet locations can be passed as xy coordinate pairs,
+        but also derived from the most downstream cell(s) within a area of interest
+        defined by a bounding box or geometry, optionally refined by stream threshold
+        arguments.
+
+        The method can be speed up by providing an additional ``bounds`` argument which
+        should contain all upstream cell. If cells upstream of the subbasin are not
+        within the provide bounds a warning will be raised. Common use-cases include:
+
+        * {'subbasin': [x, y], '<variable>': threshold}
+
+        * {
+            'subbasin': [[x1, x2, ..], [y1, y2, ..]],
+            '<variable>': threshold, 'bounds': [xmin, ymin, xmax, ymax]
+            }
+
+        * {'subbasin': /path/to/point_geometry, '<variable>': threshold}
+
+        * {'subbasin': [xmin, ymin, xmax, ymax], '<variable>': threshold}
+
+        * {'subbasin': /path/to/polygon_geometry, '<variable>': threshold}
+
+        Interbasins are similar to subbasins but are bounded by a bounding box or
+        geometry and do not include all upstream area. Common use-cases include:
+
+        * {'interbasin': [xmin, ymin, xmax, ymax], '<variable>': threshold}
+
+        * {'interbasin': [xmin, ymin, xmax, ymax], 'xy': [x, y]}
+
+        * {'interbasin': /path/to/polygon_geometry, 'outlets': true}
+    """
+    kwargs = region.copy()
+    # NOTE: the order is important to prioritize the arguments
+    options = {
+        "basin": ["basid", "geom", "bbox", "xy"],
+        "subbasin": ["geom", "bbox", "xy"],
+        "interbasin": ["geom", "bbox", "xy"],
+        "geom": ["geom"],
+        "bbox": ["bbox"],
+        "mesh": ["UgridDataArray"],
+        "grid": ["raster"],
+    }
+    kind = next(iter(kwargs))  # first key of region
+    value0 = kwargs.pop(kind)
+
+    if kind in PLUGINS.model_plugins:
+        model_class = PLUGINS.model_plugins[kind]
+        other_model = model_class(root=value0, mode="r", logger=logger)
+        value0 = other_model.region
+        kwargs = dict(geom=other_model.region)
+        kind = "geom"
+    elif kind == "mesh":
+        if not _compat.HAS_XUGRID:
+            raise ImportError("xugrid is required to read mesh files.")
+
+        if isinstance(value0, (str, Path)) and isfile(value0):
+            kwarg = dict(mesh=xu.open_dataset(value0))
+        elif isinstance(value0, (xu.UgridDataset, xu.UgridDataArray)):
+            kwarg = dict(mesh=value0)
+        elif isinstance(value0, (xu.Ugrid1d, xu.Ugrid2d)):
+            kwarg = dict(
+                mesh=xu.UgridDataset(value0.to_dataset(optional_attributes=True))
+            )
+        else:
+            raise ValueError(
+                f"Unrecognized type {type(value0)}."
+                "Should be a path, data catalog key or xugrid object."
+            )
+        kwargs.update(kwarg)
```

Comment:
There are no real tests where kind is "mesh" I believe. We should check if this is actually working.

Reply:
I agree though I don't have enough experience with `xugrid` to do this, so if we decide to keep this I think we should ask  Hélène (or someone else that has experiences with meshes) to do it.